### PR TITLE
[DOCS]: Changelog additions and improvements to RELEASE.md

### PR DESCRIPTION
**Pull request by snewcomer**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -42,244 +42,255 @@ Twitter the Crosslinking the announcement to the following Discord channels.
 
 1. Checkout the correct branch
 
-a. For the first release of a new `LTS`, create a new branch from `origin/release`
```
Comment (igorT) [inline RELEASE.md:45]: Are the whitespace changes on purpose?

**Comment by igorT** [comment]
@snewcomer I think you might be missing changes to github ci, it's still trying to run the lts 3.12 job

File: RELEASE.md (hunk 1 above)
Comment (snewcomer) [reply RELEASE.md:45]: Ya provides a bit more clear indentation.  Lmk what you think though!

PR - https://github.com/emberjs/data/blob/7cd13169f9697db85638951813914c76379bb5cb/RELEASE.md

master - https://github.com/emberjs/data/blob/master/RELEASE.md

File: yarn.lock [hunk 2]
```diff
@@ -7296,7 +7296,8 @@ eslint-module-utils@^2.6.0:
     pkg-dir "^2.0.0"
 
 "eslint-plugin-ember-data@link:./packages/unpublished-eslint-rules":
-  version "3.21.0-alpha.0"
+  version "0.0.0"
```
Comment (igorT) [inline yarn.lock:7299]: This seems like an odd change fwiw, has eslint-plugin-ember-data been version correctly? Wondering what caused this

**Comment by igorT** [comment]
Great work overall, have one small question

**Pull request merged**